core/drivers/crypto/ele/ele.c

```diff
@@ -491,33 +493,71 @@ static TEE_Result imx_ele_rng_get_trng_state(void)
 		return TEE_SUCCESS;
 }
 
-unsigned long plat_get_aslr_seed(void)
+/*
+ * Get random data from the EdgeLock Enclave.
+ *
+ * This function can be called when the MMU is off or on.
+ * virtual/physical address translation and cache maintenance
+ * is performed if needed.
+ *
+ * @buffer: data output
+ * @size: RNG data size
+ */
+static TEE_Result imx_ele_rng_get_random(uint8_t *buffer, size_t size)
 {
 	TEE_Result res = TEE_ERROR_GENERIC;
-	uint64_t timeout = timeout_init_us(10 * 1000);
+	struct imx_ele_buf rng = { };
 	struct rng_get_random_cmd {
 		uint32_t addr_msb;
 		uint32_t addr_lsb;
 		uint32_t size;
-		uint32_t crc;
 	} cmd = { };
 	struct imx_mu_msg msg = {
 		.header.version = ELE_VERSION_HSM,
 		.header.size = SIZE_MSG_32(cmd),
 		.header.tag = ELE_REQUEST_TAG,
 		.header.command = ELE_CMD_RNG_GET,
 	};
-	unsigned long aslr __aligned(CACHELINE_SIZE) = 0;
 
-	/*
-	 * This function can only be called when the MMU is off. No
-	 * virtual/physical address translation is performed, nor cache
-	 * maintenance.
-	 */
-	assert(!cpu_mmu_enabled());
+	if (!buffer || !size)
+		return TEE_ERROR_BAD_PARAMETERS;
 
-	reg_pair_from_64((uint64_t)&aslr, &cmd.addr_msb, &cmd.addr_lsb);
-	cmd.size = sizeof(aslr);
+	if (cpu_mmu_enabled()) {
+		res = imx_ele_buf_alloc(&rng, NULL, size);
+		if (res != TEE_SUCCESS)
+			return res;
+
+		cmd.addr_msb = rng.paddr_msb;
+		cmd.addr_lsb = rng.paddr_lsb;
+	} else {
+		paddr_t pa = (paddr_t)buffer;
+
+		if (!IS_ALIGNED_WITH_TYPE(pa, uint32_t))
+			return TEE_ERROR_BAD_PARAMETERS;
+
+		reg_pair_from_64((uint64_t)pa, &cmd.addr_msb, &cmd.addr_lsb);
+	}
+
+	cmd.size = (uint32_t)size;
+
+	memcpy(msg.data.u8, &cmd, sizeof(cmd));
+
+	res = imx_ele_call(&msg);
+	if (res)
+		goto out;
+
+	if (cpu_mmu_enabled())
+		res = imx_ele_buf_copy(&rng, buffer, size);
+out:
+	imx_ele_buf_free(&rng);
+
+	return res;
+}
+
+unsigned long plat_get_aslr_seed(void)
+{
+	uint64_t timeout = timeout_init_us(10 * 1000);
+	unsigned long aslr __aligned(CACHELINE_SIZE) = 0;
```

Comment:
Checkpatch seems happier with attributes before the variable label, not between the label and `= 0` affectation.
```patch
 unsigned long plat_get_aslr_seed(void)
 {
 	uint64_t timeout = timeout_init_us(10 * 1000);
-	unsigned long aslr __aligned(CACHELINE_SIZE) = 0;
+	unsigned long __aligned(CACHELINE_SIZE) aslr = 0;
 
 	/*
 	 * Check the current TRNG state of the ELE. The TRNG must be
```
It seems gcc does not care.

Reply:
oh, that was the issue. Thanks @etienne-lms
Updated pull request.